### DISCUSSION TITLE
command criteria should be able to do nots

### DISCUSSION THREAD

**Discussion by crypticC0der** [i3/i3#4934] [Ideas]
like say if i want to do somethign to all non focused windows i wanna be able to go like
`for_window [^focused] <thing>`
that would allow much more advanced customization and doesnt seem like it would be hard to do
if you can point in the direction of the files that this would involve id be happy to write the code for it

thanks

**Comment by Airblader** [comment]
For regular expression based criteria this is already possible using negative lookaheads.

> `for_window [^focused] <thing>`

Can you explain your use case for this? I don't see how `for_window` with such a criterion would be useful.

**Reply by crypticC0der** [reply]
it could realistically be used for a fair few things
i came across wanting it with how when you start a game on steam it has a little floating window saying "game loading" and it was awkward that being hidden so i thought it would be logical to change my for_window [class="steam] to for_window [class="steam" ^floating]
i have made a hacky solution in my config so my particular use case is fixed
but logical operators being able to be used in command criteria other than the default and would allow people to do things in a single line of config rather than multiple, and in some cases may avoid unnecessary IPC usage

i dont see it being a thing people need but its a small thing, that wouldnt be hard to implement and allow easier configuation

**Reply by Airblader** [reply]
> so i thought it would be logical to change my for_window [class="steam] to for_window [class="steam" ^floating]

That's just `[class="steam" tiling]`.

**Reply by crypticC0der** [reply]
wild, i went through the documentation for a solid 20 mins and didnt see that lmao
yeah, as that has a not eqivelent and with regex you can do that easily this feature is unnecicary
sorry

**Reply by shntat** [reply]
to 
crypticC0der

type into a web browser to install last compatible version of unity for vpe with unity hub 

unityhub://2021.3.0f1/6eacc8284459 

fix for the jobs error
